CSharp/Library/Microsoft.Bot.Connector.Shared/CardEx.cs

```diff
@@ -87,5 +88,276 @@ private static Attachment CreateAttachment<T>(T card, string contentType)
                 ContentType = contentType
             };
         }
+
+        /// <summary>
+        /// Gets SSML markup for this <see cref="Attachment"/>
+        /// </summary>
+        /// <param name="attachment"></param>
+        /// <returns></returns>
+        public static string GetSpeech(this Attachment attachment)
+        {
+            if (attachment.Content is ReceiptCard)
+            {
+                return GetSpeech(attachment.Content as ReceiptCard);
+            }
+            else if (attachment.Content is SigninCard)
+            {
+                return GetSpeech(attachment.Content as SigninCard);
+            }
+            else
+            {
+                return GetStandardCardSpeech((dynamic)(attachment.Content));
```

Comment:
If the cards all inherited from a common type we wouldn't have to, so yeah. Audio, Video, Hero, Thumbnail, Animation. All have same structure, but no common type.